racket-cas/racket-cas.rkt

```diff
@@ -741,103 +758,118 @@
+  (λ (stx) (syntax-parse stx [(_ u v) #'(app numerator/denominator u (? not-one? v))]))
   (λ (stx) (syntax-parse stx [(_ u v) #'(Oslash: u v)] [_ (identifier? stx) #'Oslash:])))
 
 (module+ test
-  (check-equal? (math-match 2/3 [(⊘ u v) (list u v)]) '(2 3))
-  (check-equal? (math-match '(⊕ x 2/3) [(⊘ u v) 'matched] [_ 'unmatched]) 'unmatched)
-  (check-equal? (math-match '(* x 2/3) [(⊘ u v) (list u v)]) '((* 2 x) 3))
-  (check-equal? (math-match '(* x 2.3) [(⊘ u v) 'matched] [_ 'unmatched]) 'unmatched)
-  (check-equal? (math-match (⊗ x (Expt (⊗ 2 y z) -1)) [(⊘ u v) (list u v)]) '(x (* 2 y z)))
-  (check-equal? (math-match (⊘ x y) [(⊘ u v) (list u v)]) '(x y))
-  (check-equal? (math-match (⊘ (⊗ y 3) x) [(⊘ u v) (list u v)]) '((* 3 y) x))
-  )
+  (displayln "TEST - OSLASH")
+  (check-equal? (math-match 2/3                       [(⊘ u v) (list u v)] [_ #f]) '(2 3))  
+  (check-equal? (math-match (⊘ x y)                   [(⊘ u v) (list u v)] [_ #f]) '(x y))
+  (check-equal? (math-match (⊗ x (Expt (⊗ 2 y z) -1)) [(⊘ u v) (list u v)] [_ #f]) '(x (* 2 y z)))
+  (check-equal? (math-match (⊘ (⊗ y 3) x)             [(⊘ u v) (list u v)] [_ #f]) '((* 3 y) x))
+  (check-equal? (math-match '(* 2/3 x)                [(⊘ u v) (list u v)] [_ #f]) '((* 2 x) 3)))
+
 
 (define (Quotient: u v)
   (⊘ u v))
 
 (define-match-expander Quotient
-  ; Note: This matches everything and writes it as a quotient, unless the denominator equals to 1.
-  (λ (stx) (syntax-parse stx [(_ u v) #'(app numerator/denominator u (? (λ(d)(not (equal? 1 d))) v))]))
+  ; Note: This matches everything and writes it as a quotient.
+  (λ (stx) (syntax-parse stx [(_ u v) #'(and (app numerator u) (app denominator v))]))
   (λ (stx) (syntax-parse stx [(_ u v) #'(Quotient: u v)] [_ (identifier? stx) #'Quotient:])))
 
 (module+ test
-  (check-equal? (math-match 2/3 [(Quotient u v) (list u v)]) '(2 3))
-  (check-equal? (math-match (⊕ x 2/3) [(Quotient u v) (list u v)]) '((+ 2 (* 3 x)) 3))
-  (check-equal? (math-match '(* x 2.3) [(Quotient u v) 'matched] [_ 'unmatched]) 'unmatched)
-  (check-equal? (math-match (⊘ x (⊗ 2 y z)) [(Quotient u v) (list u v)]) '(x (* 2 y z)))
-  (check-equal? (math-match (⊘ x y) [(Quotient u v) (list u v)]) '(x y))
-  (check-equal? (math-match (⊘ (⊗ y 3) x) [(Quotient u v) (list u v)]) '((* 3 y) x))
+  (displayln "TEST - QUOTIENT")
+  (check-equal? (math-match 2/3             [(Quotient u v) (list u v)] [_ #f]) '(2 3))
+  (check-equal? (math-match '(* 2.3 x)      [(Quotient u v) (list u v)] [_ #f]) '((* 2.3 x) 1))
+  (check-equal? (math-match (⊘ x (⊗ 2 y z)) [(Quotient u v) (list u v)] [_ #f]) '(x (* 2 y z)))
+  (check-equal? (math-match (⊘ x y)         [(Quotient u v) (list u v)] [_ #f]) '(x y))
+  (check-equal? (math-match (⊘ (⊗ y 3) x)   [(Quotient u v) (list u v)] [_ #f]) '((* 3 y) x))
+  ; (check-equal? (math-match (⊕ x 2/3)       [(Quotient u v) (list u v)] [_ #f]) '((+ 2 (* 3 x)) 3)) ; not simple
```

Comment:
This fails because together is not conducted.

Reply:
Yes.